Product/Production/Services/DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/docsubmission/audit/transform/DSDeferredResponseAuditTransforms.java

```diff
@@ -0,0 +1,231 @@
+    }
+
+    @Override
+    protected String getServiceEventActionCodeResponder() {
+        return DocSubmissionAuditTransformsConstants.EVENT_ACTION_CODE_SOURCE;
+    }
+
+    @Override
+    protected AuditMessageType.ActiveParticipant getActiveParticipant(UserType oUserInfo) {
+        return null;
+    }
+
+    @Override
+    protected AuditMessageType.ActiveParticipant getActiveParticipantSource(NhinTargetSystemType target,
+        String serviceName, boolean isRequesting, Properties webContextProperties) {
+
+        String ipOrHost = isRequesting ? getDSDeferredRequestInitiatorAddress(target) : getLocalHostAddress();
+
+        AuditMessageType.ActiveParticipant participant = new AuditMessageType.ActiveParticipant();
+        participant.setUserID(isRequesting ? NhincConstants.WSA_REPLY_TO
+            : getInboundReplyToFromHeader(webContextProperties));
+        if (isRequesting) {
+            participant.setAlternativeUserID(ManagementFactory.getRuntimeMXBean().getName());
+        }
+        participant.setNetworkAccessPointID(ipOrHost);
+        participant.setNetworkAccessPointTypeCode(getNetworkAccessPointTypeCode(ipOrHost));
+        participant.getRoleIDCode().add(AuditDataTransformHelper.createCodeValueType(
+            AuditTransformsConstants.ACTIVE_PARTICIPANT_ROLE_CODE_SOURCE, null,
+            AuditTransformsConstants.ACTIVE_PARTICIPANT_CODE_SYSTEM_NAME,
+            AuditTransformsConstants.ACTIVE_PARTICIPANT_ROLE_CODE_SOURCE_DISPLAY_NAME));
+        participant.setUserIsRequestor(Boolean.TRUE);
+        return participant;
+
+    }
+
+    @Override
+    protected AuditMessageType.ActiveParticipant getActiveParticipantDestination(NhinTargetSystemType target,
+        boolean isRequesting, Properties webContextProperties, String serviceName) {
+
+        String url;
+        String ipOrHost;
+
+        AuditMessageType.ActiveParticipant participant = new AuditMessageType.ActiveParticipant();
+
+        url = isRequesting ? getWebServiceUrlFromRemoteObject(getLocalHCIDNhinTarget(),
+            NhincConstants.NHINC_XDR_RESPONSE_SERVICE_NAME) : getWebServiceUrlFromRemoteObject(
+                convertToNhinTarget(getMessageCommunityId(getAssertion(), target, isRequesting)),
+                NhincConstants.NHINC_XDR_RESPONSE_SERVICE_NAME);
+        try {
+            participant.setUserID(url);
+            ipOrHost = new URL(url).getHost();
+            participant.setNetworkAccessPointID(ipOrHost);
+            participant.setNetworkAccessPointTypeCode(getNetworkAccessPointTypeCode(ipOrHost));
+        } catch (MalformedURLException ex) {
+            LOG.error("Couldn't parse the given NetworkAccessPointID as a URL: " + ex.getLocalizedMessage(), ex);
+            participant.setUserID(AuditTransformsConstants.ACTIVE_PARTICIPANT_USER_ID_SOURCE);
+            // TODO: For now, hardcode the value to localhost; need to find out if this needs to be set
```

Comment:
Not for this PR, but we need to start working through the todo items in the audit code -- or at least start making some tickets.